terms-generators.lua

```diff
@@ -777,6 +791,31 @@ local function gen_map_methods(self, key_type, value_type)
 	}
 end
 
+---@param self MapType
+---@param map { [Value]: Value }
+---@return MapValue val
+local function map_unchecked_new_fn(self, map)
+	return setmetatable({
+		_map = map,
+		is_frozen = false,
+	}, self)
+end
+
+---@param self MapType
+---@param map { [Value]: Value }
+---@return MapValue val
+local function map_new_fn(self, map)
+	local key_type, value_type = self.key_type, self.value_type
```

Comment:
Why are you creating new variables here? Just use self.key_type directly in map_set_value